libwallet/src/internal/selection.rs

```diff
@@ -496,11 +513,13 @@ where
 	let change = total - amount - fee;
 
 	// build inputs using the appropriate derived key_ids
-	for coin in coins {
-		if coin.is_coinbase {
-			parts.push(build::coinbase_input(coin.value, coin.key_id.clone()));
-		} else {
-			parts.push(build::input(coin.value, coin.key_id.clone()));
+	if include_inputs_in_sum {
+		for coin in coins {
+			if coin.is_coinbase {
+				parts.push(build::coinbase_input(coin.value, coin.key_id.clone()));
+			} else {
+				parts.push(build::input(coin.value, coin.key_id.clone()));
+			}
```

Comment:
This code suggests  a method build::input(value, key_id, is_coinbase) to replace the current tow ?!

Reply:
Similar to above, this is in the core tx `build` lib and could stand some revision (but after HF3)